hex/lib/dependabot/hex/update_checker/version_resolver.rb

```diff
@@ -68,6 +63,11 @@ def handle_hex_errors(error)
             raise Dependabot::PrivateSourceAuthenticationFailure, org
           end
 
+          if error.message.include?("Downloading public key for")
```

Comment:
For private repos fetching the public key is authenticated as well.

Reply:
I'm hoping we can get the public-key authentication dropped, but I think this will be helpful for debugging!

Reply:
Authentication requirement dropped on our end 👍